sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferClient.java

```diff
@@ -3,93 +3,274 @@
+         *
+         * @param requiresCharging {@code true} if device must be charging for the transfer to run
+         * @return Builder with provided charging requirement set
+         */
+        public Builder setRequiresCharging(boolean requiresCharging) {
+            this.requiresCharging = requiresCharging;
+            return this;
+        }
+
+        /**
+         * Sets whether device should be idle for running the transfers.
+         * The default value is {@code false}.
+         *
+         * @param requiresDeviceIdle {@code true} if device must be idle for transfers to run
+         * @return Builder with provided idle requirement set
+         */
+        @RequiresApi(23)
+        public Builder setRequiresDeviceIdle(boolean requiresDeviceIdle) {
+            this.requiresDeviceIdle = requiresDeviceIdle;
+            return this;
+        }
+
+        /**
+         * Sets the particular {@link NetworkType} the device should be in for running
+         * the transfers.
+         *
+         * @param networkType The type of network required for transfers to run
+         * @return Builder with provided network type set
+         */
+        public Builder setRequiredNetworkType(@NonNull NetworkType networkType) {
+            this.networkType = networkType;
+            return this;
+        }
+
+        /**
+         * Sets whether device battery should be at an acceptable level for running the transfers.
+         * The default value is {@code false}.
+         *
+         * @param requiresBatteryNotLow {@code true} if the battery should be at an acceptable level
+         *                              for the transfers to run
+         * @return Builder with provided battery requirement set
+         */
+        public Builder setRequiresBatteryNotLow(boolean requiresBatteryNotLow) {
+            this.requiresBatteryNotLow = requiresBatteryNotLow;
+            return this;
+        }
+
+        /**
+         * Sets whether the device's available storage should be at an acceptable level for running
+         * the transfers. The default value is {@code false}.
+         *
+         * @param requiresStorageNotLow {@code true} if the available storage should not be below a
+         *                              a critical threshold for the transfer to run
+         * @return Builder with provided storage requirement set
+         */
+        public Builder setRequiresStorageNotLow(boolean requiresStorageNotLow) {
+            this.requiresStorageNotLow = requiresStorageNotLow;
+            return this;
+        }
+
+        /**
+         * Set the {@link Executor} used by {@link TransferClient} for all its internal
+         * book keeping, which includes creating DB entries for transfer workers, querying DB
+         * for status, submitting transfer request to {@link WorkManager}.
+         *
+         * TransferClient will enqueue maximum of two command to the serialBackgroundExecutor at any time.
```

Comment:
Nit:
```suggestion
         * TransferClient will enqueue a maximum of two commands to the taskExecutor at any time.
```

Also for consistency it might be nice to rename the field to e.g. "serialTaskExecutor"

Reply:
There are a few other comments referring to `serialBackgroundExecutor` that need to be updated as well. I can push a quick fixup or you can go through and take care of it, whichever you prefer.

Reply:
sure please do, I won't say "no" to help :)